8.2 Домашнее задание Фрагменты

tunareus opened the pull request.

tunareus committed:
feat: implement ListFragment and DetailFragment for item management, update MainActivity for two-pane support

tunareus committed:
feat: integrate navigation components and update MainActivity for fragment management

vladimirlisovskij commented on app/src/main/java/com/example/myapplication/MainActivity.kt:
```diff
@@ -20,88 +26,184 @@ class MainActivity : AppCompatActivity(), ListFragment.OnItemSelectedListener {
 
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putBoolean("state_saved", true)
```
кажется это лишнее

vladimirlisovskij commented on app/src/main/java/com/example/myapplication/MainActivity.kt:
```diff
@@ -20,88 +26,184 @@ class MainActivity : AppCompatActivity(), ListFragment.OnItemSelectedListener {
+        outState.putBoolean("state_saved", true)
+    }
+
+    override fun onConfigurationChanged(newConfig: android.content.res.Configuration) {
```
а зачем тебе этот метод? он же не вызывается сейчас при перевороте устройства

https://developer.android.com/guide/topics/resources/runtime-changes#restrict-activity

vladimirlisovskij commented on app/src/main/res/navigation/nav_graph.xml:
```diff
@@ -0,0 +1,32 @@
+    app:startDestination="@id/listFragment">
+
+    <fragment
+        android:id="@+id/listFragment"
```
через tools:layout добавляем превьюхи

vladimirlisovskij commented on app/src/main/java/com/example/myapplication/MainActivity.kt:
```diff
@@ -20,88 +26,184 @@ class MainActivity : AppCompatActivity(), ListFragment.OnItemSelectedListener {
+            val navHostFragment = supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment
+            val navController = navHostFragment.navController
+            if (savedInstanceState == null) {
+                navController.navigate(R.id.listFragment)
```
можешь еще такой вариант попробовать

https://developer.android.com/guide/navigation/use-graph/safe-args

tunareus committed:
Update navigation graph and activity logic

- Add `tools:layout` attributes to fragments in the navigation graph.
- Use `let` for null handling in `MainActivity`.
- Remove unnecessary save instance state logic.
- Remove onConfigurationChanged method.
